cogs/syntax-tests.scm

```diff
@@ -173,97 +158,105 @@
 
 (define-syntax lexical-capture
   (syntax-rules ()
-                [(_) bound-x]))
+    [(_) bound-x]))
 
-(let
-  [(bound-x 'inner)]
-  (skip-compile (check-equal? "hygiene, lexical capture" (lexical-capture) 3)))
+(let ([bound-x 'inner]) (check-equal? "hygiene, lexical capture" (lexical-capture) 3))
```

Comment:
@jrvidal - lexical capture should be fixed with this PR once it is ready and cleaned up.

I think a good chunk of this will need to be reworked in order to support local syntax-rules and let-syntax, but overall I understand the macro expansion a bit more than previously.